config.go

```diff
@@ -106,6 +108,27 @@ func (c *Config) Discover() error {
 		}
```

Comment:
I think there are three levels of precedent (I ran into this in packer, also). From most to least preferred:

1. Same directory as the binary (`/usr/local/bin`)
2. Config dir (`~/.terraform/plugins`) ??
3. Builtin

I don't know of anyone who actually uses 2.

The order in the code is 2, 3, 1. It's slightly confusing but I feel doing the "fallback" approach is easier than changing the earlier code. We can change this if you prefer the priority / order in the code are the same but I think it _is_ actually working the way we want now.